sn_networking/src/record_store.rs

```diff
@@ -252,27 +287,42 @@ impl NodeRecordStore {
             let _ = metric.set(self.records.len() as i64);
         }
 
+        let (cipher, nonce_starter) = self.encryption_details.clone();
+
         let cloned_cmd_sender = self.swarm_cmd_sender.clone();
         spawn(async move {
-            let cmd = match fs::write(&file_path, r.value) {
-                Ok(_) => {
-                    // vdash metric (if modified please notify at https://github.com/happybeing/vdash/issues):
-                    info!("Wrote record {record_key:?} to disk! filename: {filename}");
-
-                    SwarmCmd::AddLocalRecordAsStored {
-                        key: r.key,
-                        record_type,
+            let mut nonce_bytes = nonce_starter.to_vec();
+            nonce_bytes.extend_from_slice(r.key.as_ref());
+            // Ensure the final nonce is exactly 96 bits long by padding or truncating as necessary
+            // https://crypto.stackexchange.com/questions/26790/how-bad-it-is-using-the-same-iv-twice-with-aes-gcm
+            nonce_bytes.resize(12, 0); // 12 (u8) * 8 = 96 bits
```

Comment:
Maybe a function `nonce(starter, key) -> GenericArray` could help keep consistency instead of repeating the logic from L144?
The `println` below on L302 looks like a leftover, should it be removed?
Otherwise looking good :)